configure.sh

```diff
@@ -13,167 +13,235 @@
-#   fi
-#   # Install GPU version
-#   echo 'Installing tensorflow .....\n'
-#   pip install tensorflow
-# fi
-
-
-
-TF_CFLAGS=( $(python -c 'import tensorflow as tf; print(" ".join(tf.sysconfig.get_compile_flags()))') )
-TF_LFLAGS="$(python -c 'import tensorflow as tf; print(" ".join(tf.sysconfig.get_link_flags()))')"
-
-
-write_to_bazelrc "build --experimental_repo_remote_exec"
-write_to_bazelrc "build --spawn_strategy=standalone"
-write_to_bazelrc "build --strategy=Genrule=standalone"
-write_to_bazelrc "build -c opt"
-write_to_bazelrc "build --cxxopt=\"-D_GLIBCXX_USE_CXX11_ABI=1\""
-write_to_bazelrc "build --cxxopt=\"-std=c++17\""
-
-# The transitive inclusion of build rules from TensorFlow ends up including
-# and building two copies of zlib (one from bazel_rules, one from the TF code
-# baase itself). The version of zlib you get (at least in TF 2.15.0) ends up
-# producing many compiler warnings that "a function declaration without a
-# prototype is deprecated". It's difficult to patch the particular build rules
-# involved, so the approach taken here is to silence those warnings for stuff
-# in external/. TODO: figure out how to patch the BUILD files and put it there.
-write_to_bazelrc "build --per_file_copt=external/.*@-Wno-deprecated-non-prototype"
-write_to_bazelrc "build --host_per_file_copt=external/.*@-Wno-deprecated-non-prototype"
-
-# Similarly, these are other harmless warnings about unused functions coming
-# from things pulled in by the TF bazel config rules.
-write_to_bazelrc "build --per_file_copt=external/com_google_protobuf/.*@-Wno-unused-function"
-write_to_bazelrc "build --host_per_file_copt=external/com_google_protobuf/.*@-Wno-unused-function"
+# --- sanity: python is importable and has TF -------------------------------
+if [[ ! -x "${PYTHON_BIN_PATH}" ]]; then
+  die "${PYTHON_BIN_PATH} not found/executable."
+fi
 
+# Ensure TF is importable from system python (user should have installed it).
+"${PYTHON_BIN_PATH}" - <<'PY' || { echo "ERROR: tensorflow not importable by chosen Python."; exit 1; }
+import tensorflow as tf
+import tensorflow.sysconfig as sc
+print("TF:", tf.__version__)
+print("include:", sc.get_include())
+print("lib:", sc.get_lib())
+PY
+
+# --- discover TF include/lib robustly --------------------------------------
+HDR="$("$PYTHON_BIN_PATH" - <<'PY'
+import tensorflow.sysconfig as sc
+print(sc.get_include())
+PY
+)"
+
+LIBDIR="$("$PYTHON_BIN_PATH" - <<'PY'
+import os, tensorflow.sysconfig as sc
+p = sc.get_lib()
+print(p if os.path.isdir(p) else os.path.dirname(p))
+PY
+)"
+
+LIBNAME="$("$PYTHON_BIN_PATH" - <<'PY'
+import os, glob, tensorflow.sysconfig as sc
+p = sc.get_lib()
+d = p if os.path.isdir(p) else os.path.dirname(p)
+cands = glob.glob(os.path.join(d,'libtensorflow_framework.so*')) \
+     or glob.glob(os.path.join(d,'libtensorflow.so*')) \
+     or glob.glob(os.path.join(d,'_pywrap_tensorflow_internal.*'))
+print(os.path.basename(cands[0]) if cands else 'libtensorflow_framework.so.2')
+PY
+)"
```

Comment:
This is going to be slow because it imports tensorflow 4 times. It's true that the imports after the first time will be faster thanks to caching by the operating system, but still, it feels like there must be a more efficient way than running Python 4 times. Also, there is a little bit of duplication of code.

What about the following?
```suggestion
tf_output=$("${PYTHON_BIN_PATH}" - <<'PY'
import sys
import os
import glob

try:
    import tensorflow as tf
    import tensorflow.sysconfig as sc
except ImportError:
    sys.exit(1)

print(sc.get_include())

lib_path = sc.get_lib()
lib_dir = lib_path if os.path.isdir(lib_path) else os.path.dirname(lib_path)
print(lib_dir)

cands = (glob.glob(os.path.join(lib_dir, 'libtensorflow_framework.so*')) or
         glob.glob(os.path.join(lib_dir, 'libtensorflow.so*')) or
         glob.glob(os.path.join(lib_dir, '_pywrap_tensorflow_internal.*')))
print(os.path.basename(cands[0]) if cands else 'libtensorflow_framework.so.2')
PY
)

if [[ $? -ne 0 ]]; then
  echo "ERROR: tensorflow not importable by Python (${PYTHON_BIN_PATH})" >&2
  exit 1
fi

{
  read -r HDR
  read -r LIBDIR
  read -r LIBNAME
} <<< "${tf_output}"
```